src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs

```diff
@@ -322,29 +348,59 @@ private void InteractiveLogin(string tenant, ActiveDirectoryServiceSettings aadS
         /// </summary>
         private void VerifySubscription()
         {
-#region
-            if (!(string.IsNullOrEmpty(this.SubscriptionId)) || (this.SubscriptionId.Equals("None", StringComparison.OrdinalIgnoreCase)))
+            #region
+            string matchedSubscriptionId = string.Empty;
+            StringBuilder sb = new StringBuilder();
+            string callerId = string.Empty;
+            string subs = string.Empty;
+
+            if (this.TokenInfo[TokenAudience.Management] != null)
             {
-                //TokenCredentials cred = this.TokenInfo[TokenAudience.Management];
+                try { callerId = this.TokenInfo[TokenAudience.Management].CallerId; } catch { }
+            }
 
+            if (!(string.IsNullOrEmpty(this.SubscriptionId)) && !(this.SubscriptionId.Equals("None", StringComparison.OrdinalIgnoreCase)))
+            {
                 List<SubscriptionInfo> subscriptionList = ListSubscriptions(this.BaseUri.ToString(), this.TokenInfo[TokenAudience.Management]);
-                string subscriptionId = subscriptionList.Where((sub) => sub.SubscriptionId.Equals(this.SubscriptionId, StringComparison.OrdinalIgnoreCase)).FirstOrDefault().SubscriptionId;
-
-                if (string.IsNullOrEmpty(subscriptionId))
+                if (subscriptionList.Any<SubscriptionInfo>())
                 {
-                    StringBuilder sb = new StringBuilder("List of subscriptions retrieved:\r\n");
-                    foreach (SubscriptionInfo subInfo in subscriptionList)
+                    var matchedSubs = subscriptionList.Where((sub) => sub.SubscriptionId.Equals(this.SubscriptionId, StringComparison.OrdinalIgnoreCase));
+                    if(matchedSubs.IsAny<SubscriptionInfo>())
+                    {
+                        matchedSubscriptionId = matchedSubs.FirstOrDefault().SubscriptionId;
```

Comment:
Yes, and the invalid operation exception is way easier to trace back to the source.  It is just good practice to use First in this instance.  Otherwise, I would suggest not doing the FISAny chanck, and simply checking if the value of FirstOrDefault is null.